packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts

```diff
@@ -50,8 +50,10 @@ export interface SageMakerCreateTrainingJobProps extends sfn.TaskStateBaseProps
 
   /**
    *  Describes the various datasets (e.g. train, validation, test) and the Amazon S3 location where stored.
+   *
+   * @default - No inputDataConfig
    */
-  readonly inputDataConfig: Channel[];
+  readonly inputDataConfig?: Channel[];
```

Comment:
This could technically be a breaking changes depending on users setup. For example, if I define the `SageMakerCreateTrainingJobProps` like 
```
const props = {
    algorithmSpecification: {
      algorithmName: 'BlazingText',
      trainingInputMode: InputMode.FAST_FILE,
    },
    inputDataConfig: [{
      channelName: 'InputData',
      dataSource: {
        s3DataSource: {
          s3Location: S3Location.fromBucket(trainingData, 'data/'),
        },
      },
    }],
    outputDataConfig: { s3OutputLocation: S3Location.fromBucket(trainingData, 'result/') },
    trainingJobName: 'mytrainingjob',
}

props.inputDataConfig.push({
    channelName: 'InputConfig',
    dataSource: {
      s3DataSource: {
        s3Location: S3Location.fromBucket(trainingData, 'XXXXXXX'),
      },
    },
  })
```
After your change, users would have linting errors and a couple setup on typescript projects is to fail build on linter errors. 

This is pretty edge case but can be possible. I also understand the fact that this is optional in CFN schema now. Let's bring it up to team discussion?

Reply:
That's a great point. Will bring it to the team to discuss.

Reply:
Linting errors would not be something we take into account when we define breaking changes. Linters are typically a matter of preference, not correctness.

In general, weakening an input type is not a breaking change. This should be fine.

Reply:
Thanks Kendra for clarification.

Reply:
Thanks Kendra for clarification.